### ISSUE TITLE
Compilation error from pip

### ISSUE THREAD

**Issue by mcocdawc** [materialsproject/pymatgen#1647]
**Describe the bug**
Compilation bug when doing `pip install pymatgen`

**To Reproduce**
1. Download and install the currently recent [Anaconda](https://repo.anaconda.com/archive/Anaconda3-2019.10-Linux-x86_64.sh).
2. Type `pip install pymatgen`
```
gcc -pthread -B /home/limanni/anaconda3/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/home/limanni/anaconda3/include/python3.7m -I/home/limanni/anaconda3/lib/python3.7/site-packages/numpy/core/include -c pymatgen/optimization/linear_assignment.c -o build/temp.linux-x86_64-3.7/pymatgen/optimization/linear_assignment.o                                                                                                                                                                
    In file included from /home/limanni/anaconda3/lib/python3.7/site-packages/numpy/core/include/numpy/ndarraytypes.h:1832:0,                                                                                                                                                  
                     from /home/limanni/anaconda3/lib/python3.7/site-packages/numpy/core/include/numpy/ndarrayobject.h:12,                                                                                                                                                     
                     from /home/limanni/anaconda3/lib/python3.7/site-packages/numpy/core/include/numpy/arrayobject.h:4,                                                                                                                                                        
                     from pymatgen/optimization/linear_assignment.c:569:
    /home/limanni/anaconda3/lib/python3.7/site-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h:17:2: warning: #warning "Using deprecated NumPy API, disable it with " "#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-Wcpp]
     #warning "Using deprecated NumPy API, disable it with " \
      ^~~~~~~
    pymatgen/optimization/linear_assignment.c: In function '__pyx_pf_8pymatgen_12optimization_17linear_assignment_16LinearAssignment___init__.isra.43':
    pymatgen/optimization/linear_assignment.c:4673:42: warning: '__pyx_v_last' may be used uninitialized in this function [-Wmaybe-uninitialized]
                   __pyx_t_45 = (__pyx_v_last + 1);
                                ~~~~~~~~~~~~~~^~~~
    pymatgen/optimization/linear_assignment.c:3115:7: note: '__pyx_v_last' was declared here
       int __pyx_v_last;
           ^~~~~~~~~~~~
    pymatgen/optimization/linear_assignment.c:3959:20: warning: '__pyx_v_j2' may be used uninitialized in this function [-Wmaybe-uninitialized]
             __pyx_t_37 = __pyx_v_j1;
             ~~~~~~~~~~~^~~~~~~~~~~~
    pymatgen/optimization/linear_assignment.c:3129:7: note: '__pyx_v_j2' was declared here
       int __pyx_v_j2;
           ^~~~~~~~~~
    pymatgen/optimization/linear_assignment.c:4694:96: warning: '__pyx_v_m' may be used uninitialized in this function [-Wmaybe-uninitialized]
                     (__pyx_v_v[__pyx_v_j1]) = (((__pyx_v_v[__pyx_v_j1]) + (__pyx_v_d[__pyx_v_j1])) - __pyx_v_m);
                                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~
    pymatgen/optimization/linear_assignment.c:3125:26: note: '__pyx_v_m' was declared here
       __pyx_t_5numpy_float_t __pyx_v_m;
                              ^~~~~~~~~
    gcc -pthread -shared -B /home/limanni/anaconda3/compiler_compat -L/home/limanni/anaconda3/lib -Wl,-rpath=/home/limanni/anaconda3/lib -Wl,--no-as-needed -Wl,--sysroot=/ build/temp.linux-x86_64-3.7/pymatgen/optimization/linear_assignment.o -o build/lib.linux-x86_64-3.7/pymatgen/optimization/linear_assignment.cpython-37m-x86_64-linux-gnu.so
    /home/limanni/anaconda3/compiler_compat/ld: /usr/lib64/gcc/x86_64-suse-linux/7/../../../../lib64/crti.o: unable to initialize decompress status for section .debug_aranges
    /home/limanni/anaconda3/compiler_compat/ld: /usr/lib64/gcc/x86_64-suse-linux/7/../../../../lib64/crti.o: unable to initialize decompress status for section .debug_aranges
    /home/limanni/anaconda3/compiler_compat/ld: /usr/lib64/gcc/x86_64-suse-linux/7/../../../../lib64/crti.o: unable to initialize decompress status for section .debug_aranges
    /home/limanni/anaconda3/compiler_compat/ld: /usr/lib64/gcc/x86_64-suse-linux/7/../../../../lib64/crti.o: unable to initialize decompress status for section .debug_aranges
    /usr/lib64/gcc/x86_64-suse-linux/7/../../../../lib64/crti.o: file not recognized: file format not recognized
    collect2: error: ld returned 1 exit status
    error: command 'gcc' failed with exit status 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /home/limanni/anaconda3/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-j94_8mxw/pymatgen/setup.py'"'"'; __file__='"'"'/tmp/pip-install-j94_8mxw/pymatgen/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-soi2u6bs/install-record.txt --single-version-externally-managed --compile Check the logs for full command output.
```
The used compiler is:
`gcc (SUSE Linux) 7.4.1 20190905 [gcc-7-branch revision 275407]`

**Expected behavior**
IMHO it would be a good idea to not have such harsh compilation flags when shipping.

**Desktop (please complete the following information):**
- NAME="openSUSE Leap"
- VERSION="15.1"

**Comment by shyuep** [comment]
Looking at the error message, it does not seem to be a compiler flag issue. What is the compiler flag that is giving you an issue? 

We do not have SuSe Linux on hand to try to debug this.

**Comment by mkhorton** [comment]
SciPy now has its own linear assignment code, perhaps we should switch to that instead of implementing our own.

**Comment by shyuep** [comment]
@mkhorton I doubt it is an issue with linear assignment. By all means, try switching to scipy and see if it works. But if the compiler flags are the issue, there are other modules that require compilation as well.

**Comment by mcocdawc** [comment]
What I said with the compilation flags is wrong.
From first glance I assumed, that you have `-maybe-uninitialized` as warning and treat warnings as errors.
I retried on Ubuntu with the same steps and it seems indeed related to SuSe Linux.
Here is the full output of my attempts
[Open_Suse_installation.txt](https://github.com/materialsproject/pymatgen/files/3788647/Open_Suse_installation.txt).

I think the problematic line is the linking step here:
```
    /home/weser/test_conda/compiler_compat/ld: /usr/lib64/gcc/x86_64-suse-linux/7/../../../../lib64/crti.o: unable to initialize decompress status for section .debug_aranges
    /home/weser/test_conda/compiler_compat/ld: /usr/lib64/gcc/x86_64-suse-linux/7/../../../../lib64/crti.o: unable to initialize decompress status for section .debug_aranges
    /home/weser/test_conda/compiler_compat/ld: /usr/lib64/gcc/x86_64-suse-linux/7/../../../../lib64/crti.o: unable to initialize decompress status for section .debug_aranges
    /home/weser/test_conda/compiler_compat/ld: /usr/lib64/gcc/x86_64-suse-linux/7/../../../../lib64/crti.o: unable to initialize decompress status for section .debug_aranges
    /usr/lib64/gcc/x86_64-suse-linux/7/../../../../lib64/crti.o: file not recognized: file format not recognized
```

**Comment by mcocdawc** [comment]
I found the reason and it has indeed nothing to do with `pymatgen`. I am sorry for the hassle.

The problem was that I used `pip` from anaconda and conda ships their own linker. Which we can also see from the path. `/home/weser/test_conda/compiler_compat/ld`.
The relevant compiler flag was `gcc [...] -B /home/weser/test_conda/compiler_compat [...]` 
If one uses the system`s `pip` without anaconda, then everything works fine.
